examples/sources/src/main/java/com/hortonworks/streamline/examples/sources/FileReaderSpout.java

```diff
@@ -0,0 +1,223 @@
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+ **/
+package com.hortonworks.streamline.examples.sources;
+
+import com.hortonworks.streamline.streams.StreamlineEvent;
+import org.apache.storm.spout.SpoutOutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.topology.base.BaseRichSpout;
+import org.apache.storm.tuple.Fields;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+public class FileReaderSpout extends BaseRichSpout {
+    protected static final Logger LOG = LoggerFactory.getLogger(FileReaderSpout.class);
+    private static final String FIRST_NAME = "firstName";
+    private static final String LAST_NAME = "lastName";
+    private transient SpoutOutputCollector spoutOutputCollector;
+    private final String path;
+    private String delimiter;
+    private String outputStream;
+
+    public FileReaderSpout (String path) {
+        this.path = path;
+    }
+
+    public FileReaderSpout withDelimiter (String delimiter) {
+        if (delimiter == null || delimiter.isEmpty()) {
+            String errorString = "Delimiter must not be empty";
+            LOG.error(errorString);
+            throw new RuntimeException(errorString);
+        }
+        this.delimiter = delimiter;
+        return this;
+    }
+
+    public FileReaderSpout withOutputStream (String outputStream) {
+        if (outputStream == null || outputStream.isEmpty()) {
+            String errorString = "Output stream must not be empty";
+            LOG.error(errorString);
+            throw new RuntimeException(errorString);
+        }
+        this.outputStream = outputStream;
+        return this;
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer outputFieldsDeclarer) {
+        outputFieldsDeclarer.declareStream(outputStream, new Fields(StreamlineEvent.STREAMLINE_EVENT));
+    }
+
+    @Override
+    public void open(Map map, TopologyContext topologyContext, SpoutOutputCollector spoutOutputCollector) {
+        LOG.info("Opening FileReaderSpout");
+        this.spoutOutputCollector = spoutOutputCollector;
+    }
+
+    @Override
+    public void nextTuple() {
+        BufferedReader br = null;
+        try {
+            String line;
+            br = new BufferedReader(Files.newBufferedReader(Paths.get(path), StandardCharsets.UTF_8));
```

Comment:
use `try(BufferedRead br = new BufferedReader(Files.newBufferedReader(Paths.get(path), StandardCharsets.UTF_8)) {` and remove finally

Reply:
Will do